src/global_distrib/mod_global_distrib_utils.erl

```diff
@@ -20,9 +20,15 @@
 -include("mongoose.hrl").
 -include("jlib.hrl").
 
+-behaviour(gen_mod).
+
+%% gen_mod callbacks
```

Comment:
I'm not sure if I like this change and the handling of this module in general. It's a bit nitpicky, but should `*_utils` really be a `gen_mod`? I get that it holds the `host_type` to extract the opts from the `gen_mod`. But since it is already a partial solution (in comparison to introducing dynamic domains to all these modules), I don't know if it wouldn't be easier to just put the host type into the persistent term inside `mod_global_distrib` itself, as it is a `gen_mod` already, and doesn't do much on startup. Of course, that would also not be an ideal/elegant solution, as `_utils` would have to "know" about this term. But in general I feel like moving everything that happens in `start` and `stop` in this module to `mod_global_distrib` would be less surprising for the reader.

Reply:
The initialization of the persistent term needs to be done before any of the dependent modules is started, so it has to be done in a module that is a dep of all of them and in cannot be done in `mod_global_distrib`, hence the module. The name might be not the best choice, but I wanted to minimize the unnecessary changes.

Reply:
Hmm... I see, didn't think about the order in which the modules are started. Makes sense 👍